doc/concepts/pipeline-concepts/index.rst

```diff
@@ -17,6 +17,29 @@ Pipeline
  and process the data. Each of these executions is
  called a job.
 
+ Pachyderm has the following special types of pipelines:
```

Comment:
I rearranged this index files a bit. Moved Spouts and Service under Pipeline and a short description for Cron.